tests/e2e/utils/auto-fsx-setup.py

```diff
@@ -0,0 +1,478 @@
+import argparse
+import boto3
+import subprocess
+import string
+import random
+import yaml
+import urllib.request
+from shutil import which
+from time import sleep
+
+
+def main():
+    header()
+
+    verify_prerequisites()
+
+    setup_iam_authorization()
+    setup_fsx_driver()
+    setup_fsx_file_system()
+    setup_fsx_provisioning()
+
+    footer()
+
+
+def write_to_file(fs_id):
+    if WRITE_TO_FILE or WRITE_TO_FILE == "True":
+        with open("fsx-config.txt", "w") as fh:
+            fh.write(fs_id)
+
+
+def rand_name(prefix):
```

Comment:
Can you use https://github.com/awslabs/kubeflow-manifests/blob/6d72c6c115a9e9b7bec89953586a3c9b982b3acc/tests/e2e/utils/utils.py#L102 ?

Reply:
Technically yes, but for now we have kept the auto script standalone thus this one method is copied over. This should change very soon in the next iteration of the scripts.

Reply:
I would recommend to reuse the code than copying entire method.

Reply:
I agree with your point and this has already been discussed in the team. However - 
1. This change is needed across all auto-scripts and is a larger refactor which is already in the plans. 
2. Making any changes to the current scripts seems like a trivial task but ends up breaking things and takes longer than expected and IMO is beyond the scope of this PR. 
3. And most importantly **the goal of these scripts is that they are meant for the user to use as standalone scripts** which will not be the case if I start re-using methods from utils. If this is not a current goal, this PR is not needed at all and the existing fixtures serve our purpose well enough. 

Based on the above, we can either - 
1. Merge this PR as is, and then follow up script refactoring across all automated scripts in a different story
2. or, close this PR until we have the bandwidth to rewrite the scripts, which is also ok since Suraj had a similar point. 

Let me know your thoughts @goswamig.

Reply:
I have removed most common code which was being duplicated. However, the script can be cleaned even further such as some of the utilities that are only needed for FSx filesystem creation like subnet etc are still in this file. We are planning a larger refactor for some of this code as discussed last week and a lot of this code is expected to change in the next iteration.